lib/spack/spack/fetch_strategy.py

```diff
@@ -385,7 +387,11 @@ def expand(self):
                 shutil.move(tarball_container, self.stage.source_path)
 
         else:
-            shutil.move(tarball_container, self.stage.source_path)
+            os.makedirs(self.stage.source_path)
```

Comment:
I'm curious.  Why is it necessary to make the directory; move individual (albeit non-hidden) entries; then remove the tarball?  The `rename` should be more efficient.  Are there potentially hidden files that should not be retained?

Reply:
Actually yeah not that I look at it again I'm not sure why. I added a test for this here so I can try switching it back.